contrib/holland-commvault/holland_commvault/commvault.py

```diff
@@ -94,7 +105,35 @@ def main():
     args.dry_run = 0
     args.no_lock = 0
     largs = args.bksets
-    if run(args, largs):
+    spool = HOLLANDCFG.lookup('holland.backup-directory')
+    status_file = '%s/%s/newest/job_%s' % (spool, args.bksets[0], args.job)
+    pid_name = 'holland_commvault_%s' % args.job
+    pid_location = '/var/run/%s.pid' % pid_name
+    try:
+        with PidFile(pid_name):
+            ret = 0
+            if run(args, largs):
+                ret = 1
+            status = open(status_file, 'w')
+            status.write(str(ret))
+            status.close()
+            return ret
+    except PidFileAlreadyLockedError:
+        pid_file = open(pid_location, 'r')
+        pid = pid_file.read()
+        pid_file.close()
+
+        logging.info("Holland (commvault agent) is already running, waiting for the pid %s", pid)
+        while os.path.isfile(pid_location):
+            sleep(5)
+        try:
```

Comment:
Can we tighten up the exception handling for this section? (The outer try/except looks fine to me.)

i.e. not wrapping as much within the try block and not catching BaseException (or Exception) unless absolutely necessary; example:

```
try:
    status = open(status_file, 'r')
    ret = int(status.read())
except IOError:
    logging.info("Holland (commvault agent) failed to open/read status file")
    return 1
else:
    status.close()
    return ret
```

or maybe:

```
try:
    with open(status_file, 'r') as status:
        ret = int(status.read())
except IOError:
    logging.info("Holland (commvault agent) failed to read status file")
    return 1
else:
    return ret
```

to give us the functionality of a `finally` statement to make sure the file handle's always closed if we can open it but not read it. We can nest another try block to give more detailed logging on the failure if you want as well.

Reply:
I've update the error handling to what you recommended here